packages/client/src/generation/utils/buildInjectableEdgeEnv.ts

```diff
@@ -50,20 +46,16 @@ function getSelectedEnvVarNames(datasources: InternalDatasource[]) {
 /**
  * Like `tryLoadEnvs` but we only retain a subset of the env vars that will be
  * used in the case of the data proxy client. And we discard the `message` prop.
- * @param envPaths to load the .env files from
  * @param envVarNames to be selected from the load
  * @returns
  */
-function loadSelectedEnvVars(envPaths: EnvPaths, envVarNames: string[]) {
-  const loadedEnv = tryLoadEnvs(envPaths, { conflictCheck: 'warn' })
+function getEmptyEnvObjectForVars(envVarNames: string[]) {
   const selectedEnv: LoadedEnv = { parsed: {} }
 
   for (const envVarName of envVarNames) {
-    const loadedEnvVar = loadedEnv?.parsed[envVarName]
-
-    // note that we willingly keep `undefined` values
-    // because we need that later in `declareInlineEnv`
-    selectedEnv.parsed[envVarName] = loadedEnvVar
+    /** note that we willingly keep `undefined` values because
+     * we need that later in {@link declareInjectableEdgeEnv} **/
+    selectedEnv.parsed[envVarName] = undefined
```

Comment:
This is very odd and I'm still unsure, even with your comment, why we are doing this!

Reply:
I'll update the comment. The answer to your question should be clarified in `declareInjectableEdgeEnv`.

Reply:
I will update the comment in the finalization PR.